### PR TITLE
Fix edge cases in extract_references

### PR DESCRIPTION
**Pull request by Jasopaum**
## Fixes
Should fix https://github.com/arkhn/warehouse-api/issues/168

## Description
Avoid some KeyErrors

## Tests
- _include unit tests still pass.
- pyfhirstore tests still pass.

### CODE REVIEW

File: src/fhirpath/engine/base.py [hunk 1]
```diff
@@ -131,51 +131,59 @@ def extract_references(self, search_param: SearchParameter) -> Dict[str, List[st
...
-                path_element._where
-                and path_element._where.type == WhereConstraintType.T2
-            ):
-                ref_target_type = ref_attribute["reference"].split("/")[0]
-                if path_element._where.value != ref_target_type:
-                    continue
-
-            if isinstance(ref_attribute, list):
+            if ref_attribute is None:
+                continue
```
Comment (simonvadee) [inline src/fhirpath/engine/base.py:184]: hurrah

File: tests/test_search.py [hunk 2]
```diff
@@ -1421,73 +1421,6 @@ def test_searchparam_type_date_period_ap(es_data, engine):
...
-
-
-def test_searchparam_summary_count(es_data, engine):
-    """Handle _summary=count
-    Search only: just return a count of the matching resources, without returning the actual matches
-    """
-    search_context = SearchContext(engine, "Patient")
-    result = Search(search_context, params=(("_summary", "count"),))()
-    assert result.entry == []
-    assert result.total == 1
```
Comment (simonvadee) [inline tests/test_search.py:1490]: why'd you remove all this ?

File: src/fhirpath/engine/base.py (hunk 1 above)
Comment (elsiehoffet-94) [reply src/fhirpath/engine/base.py:184]: 🆒

File: tests/test_search.py (hunk 2 above)
Comment (Jasopaum) [reply tests/test_search.py:1490]: It was duplicated

**Review by elsiehoffet-94 — APPROVED** [review]

**Review by simonvadee — APPROVED** [review]
